### PR TITLE
Correctly handle null ra and dec

### PR DESCRIPTION
**Pull request by lpsinger**

### CODE REVIEW

**Comment by mcoughlin** [comment]
It needs a hybrid property I think:

i.e. your expressions for python, and for sql expressions like:
x: case([and_(ra.isnot(None), dec.isnot(None)), cos(rad(ra)) * cos(rad(dec))], else_=None)
y: case([and_(ra.isnot(None), dec.isnot(None)), sin(rad(ra)) * cos(rad(dec))], else_=None)
z: case([dec.isnot(None), sin(rad(dec))], else_=None))

"is not" does not work for SQL I think.

**Comment by lpsinger** [comment]
These fields are meant to be initialized in Python, so that we do not require the SQL engine to support trigonometric functions.

File: healpix_alchemy/point.py [hunk 1]
```diff
@@ -29,20 +29,24 @@ def __init__(self, *args, ra=None, dec=None, **kwargs):
...
     x = Column(
         'x', Float, doc='Cached and precomputed Cartesian "x" coordinate',
-        **dfl(lambda ra, dec: cos(rad(ra)) * cos(rad(dec))))
+        **dfl(lambda ra, dec: cos(rad(ra)) * cos(rad(dec))
+              if ra is not None and dec is not None else None))
 
     y = Column(
         'y', Float, doc='Cached and precomputed Cartesian "y" coordinate',
-        **dfl(lambda ra, dec: sin(rad(ra)) * cos(rad(dec))))
+        **dfl(lambda ra, dec: sin(rad(ra)) * cos(rad(dec)
```
Comment (mcoughlin) [inline healpix_alchemy/point.py:37]: Should be:

        **dfl(lambda ra, dec: sin(rad(ra)) * cos(rad(dec))
              if ra is not None and dec is not None else None))

**Review by mcoughlin — APPROVED** [review]
Good point on python vs sql. Minor parentheses issue, otherwise I think ok.

File: healpix_alchemy/point.py (hunk 1 above)
Comment (lpsinger) [reply healpix_alchemy/point.py]: Good catch! Should be fixed now. Please take another look.

**Comment by mcoughlin** [comment]
I think it's ok now. Let's see if this does it!

**Comment by mcoughlin** [comment]
@lpsinger Would you consider doing a 0.2.1 type release for this PR?

**Comment by lpsinger** [comment]
Sure, but did this work?

**Comment by mcoughlin** [comment]
Yes a few tests are failing, but I think unrelated (and definitely not due to the None thing you fixed).

**Pull request merged**